pcap4j-core/src/main/java/org/pcap4j/packet/Dot11ProbeRequestPacket.java

```diff
@@ -497,137 +501,97 @@ private Dot11ProbeRequestHeader(
-      else {
-        this.supportedOperatingClasses = null;
-      }
-      if (length > 0 && rawData[offset] == HT_CAPABILITIES.value().byteValue()) {
-        this.htCapabilities = Dot11HTCapabilitiesElement.newInstance(rawData, offset, length);
-        int elemLen = htCapabilities.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.htCapabilities = null;
-      }
-      if (length > 0 && rawData[offset] == IE_20_40_BSS_COEXISTENCE.value().byteValue()) {
-        this.twentyFortyBssCoexistence
-          = Dot112040BssCoexistenceElement.newInstance(rawData, offset, length);
-        int elemLen = twentyFortyBssCoexistence.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.twentyFortyBssCoexistence = null;
-      }
-      if (length > 0 && rawData[offset] == EXTENDED_CAPABILITIES.value().byteValue()) {
-        this.extendedCapabilities
-          = Dot11ExtendedCapabilitiesElement.newInstance(rawData, offset, length);
-        int elemLen = extendedCapabilities.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.extendedCapabilities = null;
-      }
-      if (length > 0 && rawData[offset] == SSID_LIST.value().byteValue()) {
-        this.ssidList = Dot11SsidListElement.newInstance(rawData, offset, length);
-        int elemLen = ssidList.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.ssidList = null;
-      }
-      if (length > 0 && rawData[offset] == CHANNEL_USAGE.value().byteValue()) {
-        this.channelUsage = Dot11ChannelUsageElement.newInstance(rawData, offset, length);
-        int elemLen = channelUsage.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.channelUsage = null;
-      }
-      if (length > 0 && rawData[offset] == INTERWORKING.value().byteValue()) {
-        this.interworking = Dot11InterworkingElement.newInstance(rawData, offset, length);
-        int elemLen = interworking.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.interworking = null;
-      }
-      if (length > 0 && rawData[offset] == MESH_ID.value().byteValue()) {
-        this.meshId = Dot11MeshIdElement.newInstance(rawData, offset, length);
-        int elemLen = meshId.length();
-        offset += elemLen;
-        length -= elemLen;
-      }
-      else {
-        this.meshId = null;
-      }
-
-      this.vendorSpecificElements = new ArrayList<Dot11VendorSpecificElement>();
-      while (length > 0 && rawData[offset] == VENDOR_SPECIFIC.value().byteValue()) {
-        Dot11VendorSpecificElement elem
-          = Dot11VendorSpecificElement.newInstance(rawData, offset, length);
-        vendorSpecificElements.add(elem);
-        int elemLen = elem.length();
-        offset += elemLen;
-        length -= elemLen;
+      short iterCount = 0;
```

Comment:
I want the below code to be replaced with a packet factory like StaticIpV4OptionFactory for maintainability and enhanceability.